### ISSUE TITLE
Documenting all system dependencies (capabilities, mounts, etc.)

### ISSUE THREAD

**Issue by danmx** [falcosecurity/falco#1120]
**What to document**

Falco is touching critical parts of the operating system.
It would be extremely useful to have a list of system dependencies like:
- mandatory host mounts
- required Linux capabilities
- seccomp filter
- SELinux/Apparmor profiles

for:

- stand alone Falco deployment without probes
- Falco using eBPF probe
- Falco using kernel module
- probe-loader for kernel module
- probe-loader for eBPF

So we could create least privileged Falco deployments.

**Comment by danmx** [comment]
What I got so far is:
- for falco w/ eBPF to start it'll require:
  - CAP_SYS_RESOURCE
  - CAP_SYS_ADMIN
  - mounting `debugfs` in the host: `mount -t debugfs nodev /sys/kernel/debug`
  - mounting `/sys/kernel/debug` as `read-only` in the container

**Comment by fntlnz** [comment]
I agree that this will be a very important piece of our documentation once it’s done.

Moreover, I think we can be even more granular than the single capabilities by listing the specific privileged syscalls that falco needs to do, like the bpf syscall. 

Good idea @danmx - this can help a lot and can open a lot of opportunities to help harden falco and its deployments

**Comment by leodido** [comment]
Link this to #628

**Comment by leodido** [comment]
Also, I think this is a high priority task because it could clarify a lot of concerns about the security of a security tool :)

/priority high

**Comment by leodido** [comment]
Recently, we updated the docs about this matter.

See https://falco.org/docs/running

**Comment by fntlnz** [comment]
Should we close?

**Comment by danmx** [comment]
It would be great if you could narrow the capabilities instead going for `--privileged`. Not many people will use kernels `>= 5.8` any time soon.

**Comment by ylmig** [comment]
> It would be great if you could narrow the capabilities instead going for `--privileged`. Not many people will use kernels `>= 5.8` any time soon.

any progress on this? We would need that as well and we don't want to use --privileged if there are alternatives to narrow down the capabilities

**Comment by leogr** [comment]
/remove-lifecycle stale
cc @vjjmiras @therealbobo

**Comment by Andreagit97** [comment]
What do we miss here? Now the documentation should provide all the info for our drivers:
https://falco.org/docs/event-sources/kernel/#requirements

**Comment by leogr** [comment]
I believe this has been addressed. Moving the discussion to falco-website for a double check.

cc @aijamalnk @vjjmiras @Issif
